contracts/requests/WitnetRequestMalleableBase.sol

```diff
@@ -0,0 +1,374 @@
+    {
+        return (
+            /* keccak256("io.witnet.requests.malleable") */
+            0x851d0a92a3ad30295bef33afc69d6874779826b7789386b336e22621365ed2c2
+        );
+    }
+
+
+    // ================================================================================================================
+    // --- INTERNAL FUNCTIONS -----------------------------------------------------------------------------------------    
+
+    /// @dev Initializes witnessing params and template bytecode.
+    function _initialize(bytes memory _template)
+        internal
+    {
+        assert(_template.length > 0);
+        _request().template = _template;
+
+        WitnetRequestWitnessingParams storage _params = _request().params;
+        _params.numWitnesses = 2;
+        _params.minWitnessingConsensus = 51;
+        _params.witnessingCollateral = 10 ** 9;      // 1 WIT
+        _params.witnessingReward = 5 * 10 ** 5;      // 0.5 milliWITs
+        _params.witnessingUnitaryFee = 25 * 10 ** 4; // 0.25 milliWITs
+        
+        _malleateBytecode(
+            _params.numWitnesses,
+            _params.minWitnessingConsensus,
+            _params.witnessingCollateral,
+            _params.witnessingReward,
+            _params.witnessingUnitaryFee
+        );
+    }
+
+    /// @dev Serializes new `bytecode` by combining immutable template with given parameters.
+    function _malleateBytecode(
+            uint8 _numWitnesses,
+            uint8 _minWitnessingConsensus,
+            uint64 _witnessingCollateral,
+            uint64 _witnessingReward,
+            uint64 _witnessingUnitaryFee
+        )
+        internal
+        virtual
+    {
+        require(
+            _witnessingReward > 0,
+            "WitnetRequestMalleableBase: no reward"
+        );
+        require(
+            _numWitnesses >= 1 && _numWitnesses <= 127,
+            "WitnetRequestMalleableBase: number of witnesses out of range"
+        );
+        require(
+            _minWitnessingConsensus >= 51 && _minWitnessingConsensus <= 99,
+            "WitnetRequestMalleableBase: witnessing consensus out of range"
+        );
+        require(
+            _witnessingCollateral >= 10 ** 9,
+            "WitnetRequestMalleableBase: witnessing collateral below 1 WIT"
+        );
+
+        _request().bytecode = abi.encodePacked(
+            _request().template,
+            _uint64varint(bytes1(0x10), _witnessingReward),
+            _uint8varint(bytes1(0x18), _numWitnesses),
+            _uint64varint(0x20, _witnessingUnitaryFee),
+            _uint8varint(0x28, _minWitnessingConsensus),
+            _uint64varint(0x30, _witnessingCollateral)
+        );
+        _request().hash = _request().bytecode.hash();
+        emit WitnessingParamsChanged(
+            msg.sender,
+            _numWitnesses,
+            _minWitnessingConsensus,
+            _witnessingCollateral,
+            _witnessingReward,
+            _witnessingUnitaryFee
+        );
+    }
```

Comment:
As I originally commented on this piece of code, it seems rather overkill to me to serialize the whole thing only to malleate a single field.

Reply:
`_malleateBytecode(..)` can be called from three public functions: `setWitnessingCollateral(uint64)`, `setWitnessingFees(uint64,uint64)` and `setWitnessingQuorum(uint8,uint8)`. The latter two actually modify two fields, not just one. 

I agree first thought would be keep track of offset positions in which each field is placed, but it's not that simple because length of "varint" values depend on the actual value to be encoded. Checking actual length before overwritting would not preserve the need of re-concatenating the whole bytecode.